config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/keystore/KeystoreHelper.java

```diff
@@ -0,0 +1,135 @@
+        try {
+            ks.store(stream, KEYSTORE);
+        } catch (IOException | NoSuchAlgorithmException | CertificateException | KeyStoreException e) {
+            throw new KeyStoreCreationException("Error saving Key Store", e);
+        }
+        return stream.toByteArray();
+    }
+
+    private KeyStore loadKeyStore(InputStream stream, char[] password) throws CertificateException, NoSuchAlgorithmException, IOException, KeyStoreException {
+        KeyStore ks = getInstance("PKCS12");
+        ks.load(stream, password);
+
+        return ks;
+    }
+
+    private void storeKey(CertificateFactory certificateFactory, KeyStore ks, PrivateKey pk) {
+        try {
+            ks.setKeyEntry(pk.getAlias(), loadKey(pk), KEYSTORE, loadCertificateChain(pk, certificateFactory));
+        } catch (KeyStoreException e) {
+            throw new KeyStoreCreationException("Error adding Private Key '" + pk.getAlias() + "' to Key Store", e);
+        }
+    }
+
+    private Key loadKey(PrivateKey pk) {
+        File p12File = new File(pk.getPrivateKeyDirectory(), P12_FILE_NAME);
+        File txtFile = new File(pk.getPrivateKeyDirectory(), TXT_FILE_NAME);
+
+        if (p12File.exists() && txtFile.exists()) {
+            throw new KeyStoreCreationException("Can have the Private Key '" + pk.getAlias() + "' stored in p12 or txt files, but not both");
+        }
+        if (p12File.exists()) {
+            return loadKeyFromPKCS12KeyStore(pk, p12File);
+        }
+        if (txtFile.exists()) {
+            return loadKeyFromTextFile(pk, txtFile);
+        }
+
+        throw new KeyStoreCreationException("Key file for Private Key " + pk.getAlias() + " not found");
+    }
+
+    private Key loadKeyFromTextFile(PrivateKey pk, File txtFile) {
+        try (InputStreamReader reader = new InputStreamReader(Files.newInputStream(txtFile.toPath()))) {
+            PEMParser parser = new PEMParser(reader);
+            PrivateKeyInfo privateKeyInfo = (PrivateKeyInfo) parser.readObject();
+
+            return new JcaPEMKeyConverter().getPrivateKey(privateKeyInfo);
+        } catch (IOException e) {
+            throw new KeyStoreCreationException("Error loading Key file for '" + pk.getAlias() + "'", e);
+        }
+    }
+
+    private Key loadKeyFromPKCS12KeyStore(PrivateKey pk, File p12File) {
+        try (InputStream stream = Files.newInputStream(p12File.toPath())) {
+            KeyStore pks = loadKeyStore(stream, pk.getKeyPassword().toCharArray());
+            return pks.getKey(pk.getAlias(), pk.getKeyPassword().toCharArray());
+        } catch (IOException | KeyStoreException | NoSuchAlgorithmException | UnrecoverableKeyException | CertificateException e) {
+            throw new KeyStoreCreationException("Error loading Key file for '" + pk.getAlias() + "'", e);
+        }
+    }
+
+    private Certificate[] loadCertificateChain(PrivateKey privateKey, CertificateFactory certificateFactory) {
```

Comment:
We should get both the private key and cert chain from a p12 file.
For now, lets support only loading private keys from p12 files